Add support for query endpoint

alukach opened the pull request:
This is a very simple helper to allow a user to send a query to the Query Service. It supports both synchronous requests (i.e. blocking until results are available) and async requests (just getting the URL of the results.

When query is running in a synchronous fashion, if a query fails then a `QueryFailure` error will be thrown.

abarciauskas-bgse commented on maap/maap.py:
```diff
@@ -195,6 +201,64 @@ def getJobStatus(self, jobid):
+            results=True (default 180)
+        wait_interval -- number of seconds to wait between each poll for
+            results, only used if results=True (default .5)
+        """
```
💯 for documentation

alukach commented on maap.cfg:
```diff
@@ -13,6 +13,7 @@ algorithm_build = http://%(maap_host)s/api/algorithm/build
 job_status = http://%(maap_host)s/api/job/status
 wmts = http://%(maap_host)s/api/wmts
 tiler_endpoint = https://8e9mu91qr6.execute-api.us-east-1.amazonaws.com/production
+query_endpoint = http://%(maap_host)s/api/query
```
I believe that this should be `https` when deployed.

abarciauskas-bgse commented on maap/maap.py:
```diff
@@ -195,6 +201,64 @@ def getJobStatus(self, jobid):
+        src -- a dict-like object stipulating which dataset is to be queried.
+            Object must contain either a 'Granule' or 'Collection' key.
+            Collection-related value must contain 'ShortName' and 'VersionId'
+            entries. Granule-related value must contain a 'Collection' entry,
```
Does a granule make sense? I recall having the thought that it really only makes sense to query the collection (and then you can do queries on the collection that could get at data that originated from a granule file like for site name or project)

I'm thinking when developing the feature of "querying a collection via API" we discard the concept of granules.

abarciauskas-bgse commented on maap/maap.py:
```diff
@@ -195,6 +201,64 @@ def getJobStatus(self, jobid):
         )
         return response
 
+    def executeQuery(self, src, query={}, results=True, timeout=180, wait_interval=.5):
```
What do you think about adding tests for this functionality?

alukach replied:
Done (added crude integration test)

alukach commented:
@abarciauskas-bgse Alright, this should be good to go.  Changes:

* I added some logic to follow redirects when scheduling an execution. By default, `requests` follows `POST` redirects (i.e. end user accesses via `http` protocol or forgets a trailing slash) with a `GET` request. This doesn't work for our needs, so I added custom logic to handle such situations.
* Added a crude integration test.

@bsatoriu Feel free to weigh in with opinions as well.
